docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+2. `ModelToModel` - which specifies that this Schema Field should be mapped onto another Model. This happens when the Fields `ObjectDefinition` is either a `Reference`, `Dictionary[string, Reference]`, `List[Reference]` or `Set[Reference]`.
+3. `Manual` - specifies that this Field must be manually mapped (for example due to complex behaviour).
+
+> _Future thought:_ additional Field Mapping types could be added to support additional use-cases. For example, transforming values (e.g. such as having an `Integer` used in the Terraform Schema, but a `String` exposed in the SDK - or to support handling `SkuName` etc). 
+
+### a. `DirectAssignment` Field Mappings
+
+A `DirectAssignment` mapping specifies that the value of a Field within the Terraform Schema should be mapped onto the value of a Field within the related (Go) SDK Model - and vica-versa.
+
+This requires that the `TerraformObjectDefinitionType` used in the Terraform Schema Field and the `ObjectDefinitionType` used in the Field within the SDK Model are compatible - for example mapping a `String` (Schema) onto a `String` (SDK).
+
+Mappings between the two different Object Definitions are defined within the Terraform Generator, both [for simple types](https://github.com/hashicorp/pandora/blob/d893d8c6533bbfc3feab00dea5fcf58f880897df/tools/generator-terraform/generator/mappings/assignment_direct.go#L10-L22) and [complex types requiring some form of transformation](https://github.com/hashicorp/pandora/blob/d893d8c6533bbfc3feab00dea5fcf58f880897df/tools/generator-terraform/generator/mappings/assignment_direct_transform.go#L5-L14).
+
+Conversions between Object Definition Types (for example the Schema containing an Integer but the SDK containing a String value) are not supported at this time - but if needed would likely be supported as a new type of Mapping.
+
+Direct Assignment Mappings support the following properties:
+
+* `schemaModelName` - specifies the name of the Terraform Schema Model where this value should be mapped from.
+* `schemaFieldPath` - specifies the path to the Field within the Terraform Schema Model defined in `schemaModelName` which this should be mapped from.
+* `sdkModelName` - specifies the name of the SDK Model where this value should be mapped onto.
+* `sdkFieldPath` - specifies the Path to the Field within the SdkModel where the Schema Field should be mapped onto.
+
+> **Note:** Whilst `schemaFieldPath` and `sdkFieldPath` are named `Path` to allow supporting indexed keys in the future (for example using `AppSettings.SomeSetting` to retrieve the value for the key `SomeSetting` within the Dictionary defined in the `AppSettings` property) - at this point in time these only support top-level fields (e.g. `Name`).
+
+An example of a Direct Assignment Mapping can be seen below:
+
+> **Note:** A more complete example can be seen under `Examples` below.
```

Comment:
I understand this is trying to be helpful but repeating it multiple times throughout this guide is unnecessary bloat and noise for a topic that's complex. Say it once, probably at the top of the document.